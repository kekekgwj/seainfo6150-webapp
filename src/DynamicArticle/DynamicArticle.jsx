import React from "react";
import HTMLText from "../HTMLText/HTMLText";
import { Link } from "react-router-dom";
import style from "./DynamicArticle.module.css";
import ArticleImage from "../ArticleImage/ArticleImage";

const DynamicArticle = (props) => {
  const { article = {} }= props;
  
  if (!article || Object.keys(article).length === 0) {
    return (
      <p>Invalid slug</p>
    )
  }
  const { 
          title,
          author, 
          authorEmail,
          timeStamp,
          displayDate,
          text,
          image 
        } = article;
  const url = image ? image._url : null;
  return (
    <div className="DrnamicArticleWrapper">
      <div>
        <Link to="/articleList" className={style.backBtn}>BACK</Link>
      </div>
      <div className={style.articleWrapper}> 
        <article >
          <ArticleImage url={url} title={title}></ArticleImage>
          <header>
            <h1>{ title }</h1>
            <address>
              By { author }&nbsp;
              <a href={`mailto:${ authorEmail }`}>
                { authorEmail }
              </a>
              <br/>
            </address>
            <time dateTime={ timeStamp }>{ displayDate }</time>
          </header>
          <HTMLText text={ text } />
        </article>
      </div>
    </div>


  );
};

export default DynamicArticle;
