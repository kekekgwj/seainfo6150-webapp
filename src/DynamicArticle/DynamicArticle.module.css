.articleWrapper {
    margin-top: 30px;
    padding: 20px;
    border: 1px dashed red;
    border-radius: 10px;
    box-shadow: 0 0 10px 5px RGB(176,23,31,0.5);
}
.backBtn {
    color: red;
    font-weight: bold;
    font-size: 20px;
}