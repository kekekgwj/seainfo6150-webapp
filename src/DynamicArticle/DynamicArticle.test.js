import React from "react";
import DynamicArticle from "./DynamicArticle.jsx";

describe("DynamicArticle tests", () => {
    it("renders correctly", () => {
        const { container } = render(<DynamicArticle/>)
        expect(container).toMatchSnapshot();
    })
})