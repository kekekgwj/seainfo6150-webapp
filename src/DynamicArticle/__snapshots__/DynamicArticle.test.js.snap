// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicArticle tests renders correctly 1`] = `
<div>
  <p>
    Invalid slug
  </p>
</div>
`;
